src/datasources/cache/redis.cache.service.ts

```diff
@@ -70,6 +70,17 @@ export class RedisCacheService
     return result;
   }
 
+  async incrementAndGet(
+    cacheKey: string,
+    expireTimeSeconds: number | undefined,
+  ): Promise<number> {
+    const result = await this.client.incr(cacheKey);
+    if (expireTimeSeconds !== undefined && expireTimeSeconds > 0) {
+      await this.client.expire(cacheKey, expireTimeSeconds, 'NX');
+    }
+    return result;
```

Comment:
We should perform these two steps on a transaction. In the event that `INCR` is successful and `EXPIRE` is not, we are setting a key which doesn't expire ever. Suggestion:

```typescript
const transaction = this.client.multi().incr(cacheKey);
if (expireTimeSeconds !== undefined && expireTimeSeconds > 0) {
  transaction.expire(cacheKey, expireTimeSeconds, 'NX');
}
const [incrRes] = await transaction.get(cacheKey).exec();
return Number(incrRes);
```

Reply:
I do agree this would be more strict if wrapped into a transaction, but at the same time, I think it would have an associated performance cost. 

> In the event that INCR is successful and EXPIRE is not, we are setting a key which doesn't expire ever

Regarding this case, the subsequent call to the `increment` function would execute the `expire` command, and the key would have a TTL then. But it is true that if there is no subsequent call the key would live forever in the cache.

So I've added your suggestion in [66f1152](https://github.com/safe-global/safe-client-gateway/pull/1225/commits/66f1152d921315ee265df6e48f67afcb63edead5)

Thanks!